fix compilation error on fuzzers (asio deprecated related)

Fix the following compilation error:

In file included from /src/crow/tests/fuzz/template_fuzzer.cpp:4:
In file included from /src/crow/include/crow.h:7:
/src/crow/include/crow/socket_adaptors.h:47:20: error: no member named 'get_io_service' in 'asio::basic_stream_socket<asio::ip::tcp>'
   47 |             return GET_IO_CONTEXT(socket_);
      |                    ^~~~~~~~~~~~~~~~~~~~~~~
/src/crow/include/crow/socket_adaptors.h:24:32: note: expanded from macro 'GET_IO_CONTEXT'
   24 | #define GET_IO_CONTEXT(s) ((s).get_io_service())
      |                            ~~~ ^
1 error generated.

--- a/include/crow/socket_adaptors.h
+++ b/include/crow/socket_adaptors.h
@@ -18,7 +18,7 @@
 #endif
 #include "crow/settings.h"
 
-#if (CROW_USE_BOOST && BOOST_VERSION >= 107000) || (ASIO_VERSION >= 101300)
+#if (CROW_USE_BOOST && BOOST_VERSION >= 107000) || (ASIO_VERSION >= 101008)
 #define GET_IO_CONTEXT(s) ((asio::io_context&)(s).get_executor().context())
 #else
 #define GET_IO_CONTEXT(s) ((s).get_io_service())

